### DISCUSSION TITLE
Question about the kalman filters used in mrs_uav_odometry

### DISCUSSION THREAD

**Discussion by fmukwege** [ctu-mrs/mrs_uav_system#98] [General]
Hello MRS Team, 

I would like to know how does the LKF's coded in mrs_uav_odometry perform when getting data from sensor at differents rates. In my case I used a RTK as XYZ position sensor and when using "RTK" as state and altitude estimator, it is fused with the vel_baro and vel_mavros data. These last two definitely comes at higher rate than the RTK. 

My question is : "How the LKF deals with sensor data coming at different rates ?"

Best regards,

Frank

**Comment by klaxalk** [comment]
Hello, the Kalman filter implementation within our system does not rely on the sensor data coming at a specific rate. The filter's corrections are incorporated asynchronously. So to answer your question - the filter handles sensor data coming at different rates implicitly.

If you want to see the specific technical detail, just take a look at the [implementation](https://github.com/ctu-mrs/mrs_lib/blob/master/include/mrs_lib/lkf.h) :)
